src/libs/defaultRedirection/defaultRedirection.spec.ts

```diff
@@ -0,0 +1,242 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+import CozyClient, { createMockClient } from 'cozy-client'
+
+import strings from '/constants/strings.json'
+import * as DefaultRedirection from '/libs/defaultRedirection/defaultRedirection'
+import {
+  NAVIGATION_APP_SLUG,
+  DEFAULT_REDIRECTION_DELAY_IN_MS,
+  InstanceSettings,
+  fetchDefaultRedirectionUrl,
+  setDefaultRedirectionUrl,
+  getDefaultRedirectionUrl,
+  fetchAndSetDefaultRedirectionUrl,
+  fetchAndSetDefaultRedirectionUrlInBackground,
+  getOrFetchDefaultRedirectionUrl,
+  getParamsWithDefaultRedirectionUrl
+} from '/libs/defaultRedirection/defaultRedirection'
+import { formatRedirectLink } from '/libs/functions/formatRedirectLink'
+
+jest.mock('@react-native-async-storage/async-storage')
+jest.mock('/libs/functions/formatRedirectLink')
+
+const mockedFormatRedirectLink = formatRedirectLink as jest.MockedFunction<
+  typeof formatRedirectLink
+>
+
+const NAVIGATION_APP_URL = `http://${NAVIGATION_APP_SLUG}.mycozy.test/#/alan`
+const DRIVE_URL = 'http://drive.mycozy.test/#/folder'
+const CONTACT_URL = 'http://contacts.mycozy.test/#/'
```

Comment:
Do ou only test nested domain? Can you add a test for a flat one? 
```
https://contacts-foo.mycozy.cloud
```

Reply:
Right. I added tests in the function checking flat/nested.